kratos/includes/define.h

```diff
@@ -484,6 +484,159 @@ catch(...) { Block KRATOS_THROW_ERROR(std::runtime_error, "Unknown error", MoreI
     KRATOS_REGISTER_VARIABLE(name##_ZY) \
     KRATOS_REGISTER_VARIABLE(name##_ZZ)
 
+//-----------------------------------------------------------------
+//
+//  Variables time derivatives
+//
+//-----------------------------------------------------------------
+
+#ifdef KRATOS_CREATE_VARIABLE_WITH_TIME_DERIVATIVE
+#undef KRATOS_CREATE_VARIABLE_WITH_TIME_DERIVATIVE
+#endif
+#define KRATOS_CREATE_VARIABLE_WITH_TIME_DERIVATIVE(type, name, variable_derivative) \
+    /*const*/ Kratos::Variable<type > name(#name, &variable_derivative);
+
+#ifdef KRATOS_CREATE_VARIABLE_WITH_TIME_DERIVATIVE_WITH_ZERO
+#undef KRATOS_CREATE_VARIABLE_WITH_TIME_DERIVATIVE_WITH_ZERO
+#endif
+#define KRATOS_CREATE_VARIABLE_WITH_TIME_DERIVATIVE_WITH_ZERO(type, name, zero, variable_derivative) \
```

Comment:
@KratosMultiphysics/technical-committee do we need this still of are the `_WITH_ZERO` macros legacy?

Reply:
I would not add the time derivative one until we have a use case